daguerre/models.py

```diff
@@ -99,17 +109,57 @@ def delete_adjusted_images(sender, **kwargs):
     qs.delete()
 
 
+def upload_to(instance, filename):
+    """
+    Construct the directory path where the adjusted images will be saved to
+    using the MD5 hash algorithm.
+
+    Can be customized using the DAGUERRE_PATH setting set in the project's
+    settings. If left unspecified, the default value will be used, i.e. 'dg'.
+    WARNING: The maximum length of the specified string is 13 characters.
+
+    Example:
+    * Default: dg/ce/2b/7014c0bdbedea0e4f4bf.jpeg
+    * DAGUERRE_PATH = 'img': img/ce/2b/7014c0bdbedea0e4f4bf.jpeg
+
+    Known issue:
+    * If the extracted hash string is 'ad', ad blockers will block the image.
+      All occurrences of 'ad' will be replaced with 'ag' since the MD5 hash
+      produces letters from 'a' to 'f'.
+    """
+
+    first_dir = None
+    if hasattr(settings, 'DAGUERRE_PATH'):
+        first_dir = settings.DAGUERRE_PATH
+
+    if not first_dir or len(first_dir) > 13:
+        msg = ('The DAGUERRE_PATH value is more than 13 characters long! '
+               'Falling back to the default value: "dg".')
+        warnings.warn(msg)
+        first_dir = 'dg'
+
+    # Avoid TypeError on Py3 by forcing the string to bytestring
+    # https://docs.djangoproject.com/en/dev/_modules/django/contrib/auth/hashers/
+    # https://github.com/django/django/blob/master/django/contrib/auth/hashers.py#L524
+    str_for_hash = force_bytes('{} {}'.format(filename, datetime.utcnow()))
+    hash_for_dir = hashlib.md5(str_for_hash).hexdigest().replace('ad', 'ag')
```

Comment:
I've added a comment, but I've kept the more detailed explanation in the function's docstring. This way it can appear in the IDE's *show documentation* popup.